samples/subsys/nvs/README.rst

```diff
@@ -0,0 +1,23 @@
+.. _nvs-sample:
+
+NVS: Non-Volatile Storage
+#########################
+
+Overview
+********
+
+A simple application demonstrating the use of the nvs module for Non-Volatile
```

Comment:
Clean up some capitalization issues, and combine the sentences a bit for clarity:

    This is a simple application demonstrating use of the NVS
    module for non-volatile (flash) storage.  In this application,
    a counter is incremented on every reboot and stored in flash,
    the application reboots, and the reboot counter data is retrieved.